Package.swift

```diff
@@ -30,3 +23,115 @@ let package = Package(
         .testTarget(name: "SQLiteNIOTests", dependencies: ["SQLiteNIO"]),
     ]
 )
+
+#if os(macOS)
```

Comment:
I'm not either. Do you have a sense for which options we would like? Here's the diff of options that at least macOS 12.6 has but Ubuntu 20.04 does not:
```
.define("SQLITE_ATOMIC_INTRINSICS", to: "1"),
    .define("SQLITE_DEFAULT_CACHE_SIZE", to: "2000"),
    .define("SQLITE_DEFAULT_FILE_FORMAT", to: "4"),
    .define("SQLITE_DEFAULT_JOURNAL_SIZE_LIMIT", to: "32768"),
    .define("SQLITE_DEFAULT_LOOKASIDE", to: "1200,102"),
    .define("SQLITE_DEFAULT_MEMSTATUS", to: "0"),
    .define("SQLITE_DEFAULT_MMAP_SIZE", to: "0"),
    .define("SQLITE_DEFAULT_PAGE_SIZE", to: "4096"),
    .define("SQLITE_DEFAULT_PCACHE_INITSZ", to: "20"),
    .define("SQLITE_DEFAULT_SECTOR_SIZE", to: "4096"),
    .define("SQLITE_DEFAULT_SYNCHRONOUS", to: "2"),
    .define("SQLITE_DEFAULT_WAL_AUTOCHECKPOINT", to: "1000"),
    .define("SQLITE_DEFAULT_WAL_SYNCHRONOUS", to: "1"),
    .define("SQLITE_DEFAULT_WORKER_THREADS", to: "0"),
    .define("SQLITE_ENABLE_LOCKING_STYLE", to: "1"),
    .define("SQLITE_MALLOC_SOFT_LIMIT", to: "1024"),
    .define("SQLITE_MAX_ATTACHED", to: "10"),
    .define("SQLITE_MAX_COLUMN", to: "2000"),
    .define("SQLITE_MAX_COMPOUND_SELECT", to: "500"),
    .define("SQLITE_MAX_DEFAULT_PAGE_SIZE", to: "8192"),
    .define("SQLITE_MAX_EXPR_DEPTH", to: "1000"),
    .define("SQLITE_MAX_FUNCTION_ARG", to: "127"),
    .define("SQLITE_MAX_LENGTH", to: "2147483645"),
    .define("SQLITE_MAX_LIKE_PATTERN_LENGTH", to: "50000"),
    .define("SQLITE_MAX_MMAP_SIZE", to: "1073741824"),
    .define("SQLITE_MAX_PAGE_COUNT", to: "1073741823"),
    .define("SQLITE_MAX_PAGE_SIZE", to: "65536"),
    .define("SQLITE_MAX_SQL_LENGTH", to: "1000000000"),
    .define("SQLITE_MAX_TRIGGER_DEPTH", to: "1000"),
    .define("SQLITE_MAX_VARIABLE_NUMBER", to: "500000"),
    .define("SQLITE_MAX_VDBE_OP", to: "250000000"),
    .define("SQLITE_MAX_WORKER_THREADS", to: "8"),
    .define("SQLITE_STMTJRNL_SPILL", to: "131072"),
    .define("SQLITE_TEMP_STORE", to: "1"),
    .define("SQLITE_THREADSAFE", to: "2"),
    .define("SQLITE_BUG_COMPATIBLE_20160819"),
    .define("SQLITE_DEFAULT_AUTOVACUUM"),
    .define("SQLITE_DEFAULT_CKPTFULLFSYNC"),
    .define("SQLITE_DEFAULT_RECURSIVE_TRIGGERS"),
    .define("SQLITE_ENABLE_API_ARMOR"),
    .define("SQLITE_ENABLE_BYTECODE_VTAB"),
    .define("SQLITE_ENABLE_DBPAGE_VTAB"),
    .define("SQLITE_ENABLE_EXPLAIN_COMMENTS"),
    .define("SQLITE_ENABLE_NORMALIZE"),
    .define("SQLITE_ENABLE_SNAPSHOT"),
    .define("SQLITE_ENABLE_STMT_SCANSTATUS"),
    .define("SQLITE_MUTEX_UNFAIR"),
    .define("SQLITE_OMIT_AUTORESET"),
    .define("SQLITE_OMIT_LOAD_EXTENSION"),
    .define("SQLITE_SYSTEM_MALLOC"),
```
Notably, macOS uses `SQLITE_THREADSAFE=2` as a default but Ubuntu uses `SQLITE_THREADSAFE=1`. I don't think this matters much since we manually set it to `1 (serial)` in the library. But some of the others I'm unsure about, such as `SQLITE_SYSTEM_MALLOC`, `SQLITE_OMIT_LOAD_EXTENSION`, `SQLITE_MUTEX_UNFAIR` and some are just plain strange, i.e. missing from SQLite's documentation, i.e. `SQLITE_BUG_COMPATIBLE_20160819`. I have a hunch that Apple provides their own, modified version of SQLite...

And for completeness, here are the options that Ubuntu has but macOS does not:
```
.define("SQLITE_MAX_SCHEMA_RETRY", to: "25"),
    .define("SQLITE_MAX_VARIABLE_NUMBER", to: "250000"),
    .define("SQLITE_THREADSAFE", to: "1"),
    .define("SQLITE_ENABLE_LOAD_EXTENSION"),
    .define("SQLITE_ENABLE_STMTVTAB"),
    .define("SQLITE_ENABLE_UNLOCK_NOTIFY"),
    .define("SQLITE_LIKE_DOESNT_MATCH_BLOBS"),
    .define("SQLITE_OMIT_LOOKASIDE"),
    .define("SQLITE_SECURE_DELETE"),
    .define("SQLITE_SOUNDEX"),
```

I haven't yet checked on all of them to give a proper recommendation, but curious what your thoughts are.

Reply:
Apple does indeed provide a modified SQLite, for various reasons (whose validity varies in turn), but for our purposes we can ignore it, fortunately.

Most of the extra options macOS specifies are performance tuning (probably heavily in favor of Core Data's use case at the expense of others, knowing Apple), Omitting the extension loading functionality is a security measure; "MUTEX_UNFAIR" and "CKPTFULLFSYNC" are platform configuration that should be autodetected. In general, I think most of the options that show up on macOS are the result of specifying explicit values for things that would've been the default anyway.

For the most part we should use the Ubuntu configuration when it comes to feature enables and things like max limits; on macOS we should include those of the platform tuning options that are available in the embedded version (things like `BUG_COMPATIBLE_20160819` aren't an issue for us - that one's just a signal that Apple's version is emulating a old bug in the library for compatibility, which we don't want to do anyway).

Reply:
I have adjusted the compile options to target Ubuntu 22.04 (Jammy) for all platforms. I kept some of the tuning options on macOS but did remove many that appeared to be SQLite defaults. Let me know if you would like me to restore any.